src/Compilers/CSharp/Portable/Symbols/Source/SourceNamespaceSymbol.cs

```diff
@@ -314,13 +314,11 @@ internal override NamespaceExtent Extent
             // NOTE: a name maps into values collection containing types only instead of allocating another
             // NOTE: array of NamedTypeSymbol[] we downcast the array to ImmutableArray<NamedTypeSymbol>
 
-            var builder = new NameToSymbolMapBuilder(_mergedDeclaration.Children.Length);
-            foreach (var declaration in _mergedDeclaration.Children)
-            {
-                builder.Add(BuildSymbol(declaration, diagnostics));
-            }
-
-            var result = builder.CreateMap();
+            var result = _mergedDeclaration.Children.ToDictionary(
+                static (declaration, tuple) => tuple.@this.BuildSymbol(declaration, tuple.diagnostics),
+                static symbol => symbol.Name,
+                (@this: this, diagnostics),
+                downcastUnused: (NamedTypeSymbol)null);
```

Comment:
now using common helper shared with SourceMemberContainerSymbol for the same purpose.